portals/application/modules/provision/controllers/IndexController.php

```diff
@@ -33,70 +24,56 @@ public function indexAction()
 
     public function templateAction()
     {
+        $isHttps = isset($_SERVER['HTTPS']) && !empty($_SERVER['HTTPS']);
+        $terminalUrl = $this->getRequest()->getParam('requested_url');
+        $path = $this->_getFilePath();
+        $terminalModelMapper = new \IvozProvider\Mapper\Sql\TerminalModels();
+        $terminalModel = $this->_searchGenericPattern($terminalModelMapper, $terminalUrl);
+
+        if ($terminalModel) {
+            if ($isHttps) {
+                return $this->_error(403);
+            }
+            return $this->_renderPage('generic', $path, $terminalModel);
+        }
+
+        if (!$isHttps) {
+            return $this->_error(403);
+        }
```

Comment:
Maybe It's not common for PHP programing, but won't be handy for the future yourself to add a comment here to clarify that generic requests MUST NOT use https while specific requests MUST use https?

Just a comment, feel free to ignore this suggestion, maybe I'm thinking too much `C` mode :smile: .

        if ($terminalModel) {
            // Generic Template requests must be served over HTTP
            if ($isHttps) {
                return $this->_error(403);
            }
            return $this->_renderPage('generic', $path, $terminalModel);
        }

        // Specific Template requests must be served over HTTPS
        if (!$isHttps) {
            return $this->_error(403);
        }